### DISCUSSION TITLE
How close is autofree to completeness?

### DISCUSSION THREAD

**Discussion by qpwo** [vlang/v#13588] [General]
That last 10% which needs garbage collection right now, how hard would it be to improve autofree to cover those cases?

See also: https://github.com/vlang/v/discussions/12343

**Comment by qpwo** [comment]
I am curious if it is even possible

**Reply by ghost** [reply]
I am also very interested to see if autofree will be completed.
There hasn't been a fix committed for autofree for a while now, so I don't know what's going on in development.
There have been quite a few memory management bugs reported, but none have been fixed.
You may want to list the issues.

**Reply by ylluminate** [reply]
I believe @medvednikov has temporarily shelved this since the temporary GC usage (`-gc boehm`) has resolved a lot of issues surrounding this for nearly everyone such that he can wrap up work on `c2v`. I believe we'll see a return to this after we see a public release of `c2v` at some level and perhaps some work in maybe some other areas. I know that these particular two areas are very important to Alex and the reason he's been working on them personally, in addition to many other things behind the scenes, even though they are seemingly moving at a labored pace.

**Reply by qpwo** [reply]
The pace is astounding don't get me wrong
